### ISSUE TITLE
Account for the time spread in the photon production in the scintillator

### ISSUE THREAD

**Issue by cmantill** [LDMXAnalysis/ldmx-sw#1350]
The scintillator relaxes over a finite time, so the photons arrive over a range of times, even if they are produced at the same point in the bar. This should account for effects of the scintillator bar itself and the wavelength shifter (Y-11).

Jeremy's suggestion:
1. for each energy deposition, use a PE/edep factor to determine the number of PE actually produced (include attenuation effects). At this point, one would naturally include a Poisson trial given the small number of PE involved.
2. Get the relaxation curve from Mu2e: Should be the curve representing the distribution of arrival times at the SiPM for an energy deposition at a finite point in the bar. (hopefully includes both effects from the scintillator and WLS).
3. Use the relaxation curve to pick times for each of the photo electrons to arrive.
4. Use a voltage/PE factor to covert to voltage. [for bonus points, add factor to smear gain on a photon-by-photon basis]
5. Add all the other time factors (sample phase/z-position correction, delay along the bar)
6. Send in these photoelectron-derived entries to the common digitizer code as if they were the simhits.

**Comment by tomeichlersmith** [comment]
The timing of the different pulses (and how they are digitized) is _intimately_ connected to how the HgcrocEmulator handles timing. The current version of the emulator on ldmx-sw/trunk is a good first pass, but it has several flaws highlighted in https://github.com/LDMX-Software/ldmx-sw/issues/946 .

I intentionally wanted to separate the subsystem-specific stuff (steps 1 through 5 above) and the shared stuff (the digitization done by the chip step 6). The steps above won't really change as I improve the emulator, but you probably won't get very realistic (or even helpful) digitized outputs until after I update the emulator to effectively handle pulses separated by a good amount of time. This is complicated by the fact that _if_ a bar has multiple hits in the HCal, they are _probably_ near in space and time, so these bugs would only be seen rarely in simulation. For this reason, I would encourage you to add to your to-do list expanding the digi-test to include multiple-sim-hit reconstruction that makes sure hits separated by space or time or both behaves properly.

**Long story short**: Don't worry about (or even test) the output digis until after I've updated the emulator to be better. You can still work on getting steps 1 through 5 to behave realistically, but step 6 is where my bugs come in.

**Comment by cmantill** [comment]
Ok will do: 
- Expand HcalDigiPipeline test to include multiple sim-hit reconstruction (handle hits separated by space or time, or both)

**Comment by tvami** [comment]
@tomeichlersmith @cmantill is this still an open issue?

**Comment by cmantill** [comment]
Yes, I think that we could use Ralf's input on how easy is this to implement here.
